Source/Dafny/Compilers/SinglePassCompiler.cs

```diff
@@ -1670,8 +1671,19 @@ protected class NullClassWriter : IClassWriter {
         }
       }
       if (!m.Ins.TrueForAll(f => f.IsGhost)) {
-        reason = "the method has non-ghost parameters";
-        return false;
+        var nonGhostFormals = m.Ins.Where(f => !f.IsGhost).ToList();
```

Comment:
Just a curiosity question: is this change related to the CLI arguments or is it a fix that is piggy-backing along?

Reply:
Related to the CLI  arguments? What do you mean?

Reply:
This PR is about support for CLI arguments for main in Dafny programs. I don't understand the relationship of this change to that goal.

Reply:
Oh ok. So yes this change relates to the CLI arguments.
This change in the context of resolving the "main" method which has some specificities. Previously, a main method could accept only ghost parameters, an no non-ghost parameters.
Now, a main method should be able to receive at most one non-ghost parameter, which should be an `array<string>`. All this code is there to ensure it.